internal/util/telemetry/telemetry.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetry
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"runtime"
+	"time"
+
+	"github.com/AlekSi/pointer"
+	"go.uber.org/zap"
+
+	"github.com/FerretDB/FerretDB/internal/util/state"
+	"github.com/FerretDB/FerretDB/internal/util/version"
+)
+
+// request represents telemetry request.
+type request struct {
+	Version string `json:"version"`
+	Commit  string `json:"commit"`
+	Branch  string `json:"branch"`
+	Dirty   bool   `json:"dirty"`
```

Comment:
What is `Dirty` for?

Reply:
To check if someone encounters weird problem with non-clean build with modified source. In that case our response could be "please try an official build"